java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -0,0 +1,476 @@
+        break;
+      default:
+    }
+
+    checkerDispatcher.executeCheckPostStatement(tree);
+    if (tree.parent().is(Tree.Kind.EXPRESSION_STATEMENT)) {
+      programState = ProgramState.unstack(programState, programState.stack.size()).a;
+    }
+  }
+
+  private void setSymbolicValueOnFields(MethodInvocationTree tree) {
+    if (isLocalMethodInvocation(tree)) {
+      resetFieldValues(tree);
+    }
+  }
+
+  private static boolean isLocalMethodInvocation(MethodInvocationTree tree) {
+    ExpressionTree methodSelect = tree.methodSelect();
+    if (methodSelect.is(Tree.Kind.IDENTIFIER)) {
+      return true;
+    } else if (methodSelect.is(Tree.Kind.MEMBER_SELECT)) {
+      MemberSelectExpressionTree memberSelectExpression = (MemberSelectExpressionTree) methodSelect;
+      ExpressionTree target = memberSelectExpression.expression();
+      if (target.is(Tree.Kind.IDENTIFIER)) {
+        IdentifierTree identifier = (IdentifierTree) target;
+        return THIS_SUPER.contains(identifier.name());
+      }
+    }
+    return false;
+  }
+
+  private void resetFieldValues(MethodInvocationTree tree) {
+    boolean changed = false;
+    Map<Symbol, SymbolicValue> values = Maps.newHashMap(programState.values);
+    for (Map.Entry<Symbol, SymbolicValue> entry : values.entrySet()) {
+      Symbol symbol = entry.getKey();
+      if (isField(symbol)) {
+        VariableTree variable = ((Symbol.VariableSymbol) symbol).declaration();
+        if (variable != null) {
+          changed = true;
+          SymbolicValue nonNullValue = constraintManager.supersedeSymbolicValue(variable);
+          values.put(symbol, nonNullValue);
+        }
+      }
+    }
+    if (changed) {
+      programState = new ProgramState(values, programState.constraints, programState.visitedPoints, programState.stack);
+    }
+  }
+
+  private static boolean isField(Symbol symbol) {
+    return symbol.isVariableSymbol() && !symbol.owner().isMethodSymbol();
+  }
+
+  private void logState(MethodInvocationTree mit) {
+    if (mit.methodSelect().is(Tree.Kind.IDENTIFIER) && "printState".equals(((IdentifierTree) mit.methodSelect()).name())) {
+      debugPrint(((JavaTree) mit).getLine(), node);
+    }
+  }
+
+  private void debugPrint(Object... toPrint) {
+    if (DEBUG_MODE_ACTIVATED) {
+      LOG.error(Joiner.on(" - ").join(toPrint));
+    }
+  }
+
+  public void enqueue(ExplodedGraph.ProgramPoint programPoint, ProgramState programState) {
+    int nbOfExecution = programState.numberOfTimeVisited(programPoint);
+    if (nbOfExecution > MAX_EXEC_PROGRAM_POINT) {
+      debugPrint(programState);
+      return;
+    }
+    Multiset<ExplodedGraph.ProgramPoint> visitedPoints = HashMultiset.create(programState.visitedPoints);
+    visitedPoints.add(programPoint);
+    ExplodedGraph.Node node = explodedGraph.getNode(programPoint, new ProgramState(programState.values, programState.constraints, visitedPoints, programState.stack));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "node" which hides the field declared at line 83. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AHiddenFieldCheck)